### ISSUE TITLE
feat(youtube): AI translated voiceover / Port Voice Over Translation userscript

### ISSUE THREAD

**Issue by Metadorius** [ReVanced/revanced-patches#580]
### Type

Functionality

### Issue

ReVanced YouTube app lacks AI voiceover feature which is available using userscripts on desktop.

### Feature

Implement/port Voice Over Translation userscript which uses Yandex services under the hood in order to translate the voice (in YT, in particular, but not limited to) from one language to another. You can see more details here: https://github.com/ilyhalight/voice-over-translation/blob/master/README-EN.md (contrary to the readme the plugin can actually translate either way, not just to Russian).

We've asked the author @ilyhalight in https://github.com/ilyhalight/voice-over-translation/issues/187 but he doesn't know Kotlin, although he also admits it would be great to have this feature integrated in ReVanced.

### Motivation

Despite English being top-1 international language, there are still many people who don't know English, and also there's a lot of non-English videos. It's considerably more tedious to watch videos in foreign language for people who don't know the language. For instance, I do, but many of my local friends and family members don't know and they would still benefit a lot from that, like my mom who only has her phone and can't use the userscript to translate the video). It would help greately to have this integrated in ReVanced to make videos more accessible worldwide.

**Comment by lynrayy** [comment]
Still in demand

**Comment by oSumAtrIX** [comment]
Over time, a lot of feature requests have been made, which is why I am closing some according to the [contribution guidelines](https://github.com/ReVanced/revanced-patches/blob/main/CONTRIBUTING.md#-submitting-a-feature-request):

> Requests can be accepted or rejected at the discretion of maintainers of ReVanced Patches

This feature request has been closed for one or more of the following reasons:

- Stale for too long
- Low priority
- Maintainers un-interested
- Too complex to realize

Regardless, anyone who comes across this feature request can PR it. Thanks for the feature request!
